processor/src/test/java/org/mapstruct/ap/test/value/string2enum/StringToEnumMappingTest.java

```diff
@@ -53,7 +54,20 @@ public void testRemainingAndNull() {
                     "type String to an enum type." )
         }
     )
-    public void shouldRaiseErrorWhenUsingAnyRemaining() {
+    public void shouldRaiseWarningWhenNotUsingAnyRemainingOrAnyUnmapped() {
```

Comment:
Perhaps it is my misunderstanding.. Lets see if my assumption is correct:

* MapStruct will continue to do name based mapping for the remaining properties based on tartget
* And then still raise a warning (that cannot be suppressed)?

Or.. is this the default behavior and what should the user do to force MapStruct to continue with name-based mappings based on target without issuing a warning?

Reply:
Yes your assumptions are correct, this is what happens with this PR. Suppressing the warning could be done if we implement https://github.com/mapstruct/mapstruct/issues/2339.

Before, we generated a warning and didn't do name based mapping. This in my opinion is not correct and is misleading, since a warning should not stop the name based mapping.

Reply:
The warning possibly stems from the enum-to-enum mapping, where it totally makes sense and is in line with the normal `@Mapping` (when you miss targets). Here however, it feels a bit awkward. Perhaps we should advance #2339 as part of 1.5 to fully finish this one.

Reply:
But than it has to be something like: `@ValueMapping(target = MappingConstants.THROW_EXCEPTION, source =  MappingConstants.ANY_UNMAPPED)` (all unmapped) and and `@ValueMapping(target = MappingConstants.THROW_EXCEPTION, source =  MappingConstants.ANY_UNY_REMAINING)` (all non-name-based unmapped)

Reply:
I agree, and I also like #2339, so it will be part of 1.5 for sure :). Btw the `THROW_EXCEPTION` doesn't have to be only for the `ANY_UNMAPPED` or `ANY_REMAINING`, you can also define it for a specific mapping as well (although we need to see how the inverse works there (most likely it should be ignored).